recipes/antlr4-cppruntime/all/test_package/CMakeLists.txt

```diff
@@ -12,4 +9,4 @@ if(TARGET antlr4_shared)
 else()
     target_link_libraries(${PROJECT_NAME} PRIVATE antlr4_static)
 endif()
-set_target_properties(${PROJECT_NAME} PROPERTIES CXX_STANDARD 17 CXX_STANDARD_REQUIRED ON)
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
```

Comment:
This could be a cause of error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in test_package.obj
ref:
https://stackoverflow.com/questions/14714877/mismatch-detected-for-runtimelibrary

Reply:
I think the linker error might be related to this PR: [Windows specific fixes](https://github.com/antlr/antlr4/pull/3678).
The CI result shows that all tests passed on Windows with antlr 4.11.x.

Reply:
Now I am curious about why this error didn't show up when the recipe was upgraded to support 4.10.x last time and what the best way to solve this error this time is (i.e. modifying the Conan file versus backporting the fix). This linker error showed up after I merged @uilianries's PR that made the recipe compatible with Conan 2.0, so maybe he can shed some lights on this issue.

Reply:
You talk about DropLibUuid-4.9.3 patchset ? That was @uilianries change ?

Reply:
Nothing to do with Conan 2.0 neither libuuid. Please, look into upstream CMakelists and check for static runtime forced flags. It will require a patch and if already available, please, add to conandata.yml all information.